src/modules/Dashboard/components/ActivityAssignDrawer/ActivityAssignDrawer.tsx

```diff
@@ -386,12 +386,12 @@ export const ActivityAssignDrawer = ({
             hasActivity: activityIds.length || flowIds.length,
             isTeamMember,
           });
+        } else if (activityIds.length || flowIds.length) {
+          addBanner('ActivityAutofillBanner');
         } else if (assignments[0]?.targetSubjectId) {
           addBanner('SubjectAutofillBanner', {
             hasActivity: activityIds.length || flowIds.length,
           });
-        } else if (activityIds.length || flowIds.length) {
-          addBanner('ActivityAutofillBanner');
```

Comment:
This isn't quite doing the trick. The issue as reported in the bug description is that the message `The Participant & Activity have been auto-filled. 1 row needs attention to continue.` is what's desired when there is an activity/flow defined, _and_ a target subject. Your change above actually causes the message `Your Activity was auto-filled, add Respondents to continue.` to be shown under this condition now, which isn't quite right. So I think the above change should be reverted.

Looking at the code as it was before the change, that condition _should_ have be captured by the `addBanner('SubjectAutofillBanner'…)` call above, with `hasActivity: true`. But if you dig into `SubjectAutofillBanner`, you'll see that it's not even handling the `hasActivity` prop at all – that's a flop on my part!

That component just needs to be tweaked so that the `subjectActivityAutofill` language key is used when `hasActivity` is true, and otherwise behave the same as before:

```tsx
      {hasActivity ? t('subjectActivityAutofill') : t('subjectAutofill')}
```

Reply:
Thanks for catching that, I did get those mixed up.